### PR TITLE
Faster plugin updater

### PR DESCRIPTION
**Pull request by brodock**
Changed Faraday adapter to use a persistent connection to prevent TCP/SSL handshaking at every request.

This reduced total time from ~26 min down to ~6:40 min. This can be further reduced by doing parallel requests with typhoeus.

I've also added a progress bar, and added a Rakefile to run the whole thing.

### CODE REVIEW

**Comment by brodock** [comment]
@jimmycuadra please let me know if you need any change to get this merged :)

**Comment by jimmycuadra** [comment]
I was on vacation, but I'll be looking at this PR and your other one this week. Thanks for the contribution!

File: Rakefile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+$:.unshift File.expand_path("../lib", __FILE__)
```
Comment (jimmycuadra) [inline Rakefile]: Is there a way to leverage Bundler to add this directory to the load path? Manually modifying `$:` feels a little sketchy.

File: deploy.sh [hunk 2]
```diff
@@ -4,9 +4,7 @@ set -euxo pipefail
 
 if [ -z "${SYNC_ONLY:-}" ]; then
   bundle install --path /var/bundle --jobs $(nproc) --clean
-
-  mkdir -p plugin_data
-  bundle exec ruby -Ilib -e 'require "plugin_updater"; PluginUpdater.update'
+  rake update_plugins
```
Comment (jimmycuadra) [inline deploy.sh]: Let's `bundle exec` this command.

File: lib/plugin_updater.rb [hunk 3]
```diff
@@ -43,6 +44,13 @@ def attributes_for(name)
       STDERR.puts e
     end
 
+    def rubygems_api
+      @conn ||= Faraday.new(url: RUBYGEMS_URL) do |faraday|
+        # Use the net-http-persistent implementation
```
Comment (jimmycuadra) [inline lib/plugin_updater.rb]: No need for this comment—it's obvious what the line is doing.

**Comment by jimmycuadra** [comment]
This is a very nice improvement. With a few small things addressed it should be good to merge.

**Comment by brodock** [comment]
@jimmycuadra I've addressed your suggestions, it should be ready to merge :)

**Comment by jimmycuadra** [comment]
Thanks so much!

**Pull request merged**